### PR TITLE
[WIP] add agent_radius to aabb obstacles

### PR DESCRIPTION
**Pull request by nickydonna**
Testing `agent_radius` 

Related to #50

### CODE REVIEW

**Comment by nickydonna** [comment]
@mockersf here I did a bit of a test, with the primitives, and added controls to the demo, and it seems to work fairly well

File: src/obstacles/primitive.rs [hunk 1]
```diff
@@ -152,7 +149,7 @@ impl ObstacleSource for PrimitiveObstacle {
                     copypasta::arc_2d_inner(
                         0.0,
                         std::f64::consts::PI,
-                        primitive.radius,
+                        primitive.radius + agent_radius,
```
Comment (nickydonna) [inline src/obstacles/primitive.rs:152]: Here I'm unsure if we need to add it to both calculation in the Capsule

**Comment by nickydonna** [comment]
Just pushed support for avian3d, but is pretty slow. I'm trying to add `expand` to `parry` Trimesh, which should be faster, but right now, the example timesout with Direct update

**Comment by mockersf** [comment]
Looks good, thanks!

For primitives, could you check for `CircularSector` and `CircularSegment`? When I tried they required adjustments.

On my laptop the `auto_navmesh_avian3d` is working correctly on your PR, was it already timeouting before your changes?

**Comment by mockersf** [comment]
I added agent radius on the navmesh in #59

**Comment by nickydonna** [comment]
Thanks, sorry for not been able to fllow up on this,

**Comment by mockersf** [comment]
no worries!

I've been poked by someone else to do the second idea I spoke about in #50, and turns out it was easier that I thought!

**Pull request closed without merging**